### DISCUSSION TITLE
OPA Error

### DISCUSSION THREAD

**Discussion by atharva-pb** [walt-id/.github#53] [Help] [answered]
Hi, when I'm trying to verify a dynamic policy, I'm getting the below **error**.

*ERROR id.walt.auditor.Auditor - DynamicPolicy: java.lang.IllegalStateException: Executable for OPA policy engine not installed. See https://www.openpolicyagent.org/docs/#running-opa*

Upon following the steps given in the OPA link, the error still persists even though:
1. The same steps done as in the following video (stuck at the **timestamp 14:05 - https://youtu.be/mue4UjzOZ3Q?t=845**) - Both local and docker
2.  sample OPA eval expression is working fine (**./opa eval "1*2+3"**) 
3. **./opa** **run** is opened in a parallel terminal and a Dynamic Policy is tried to be verified. 

Can someone please suggest me the correct way to use OPA if I have to verify a dynamic policy?

Thanks!

![image](https://github.com/walt-id/.github/assets/129721629/b1331420-5d77-42a8-9002-43fbf94934c1)

**Comment by severinstampler** [comment, marked as answer]
Hi @atharva-pb 
Please make sure the `opa` executable is in your `PATH` and opa is executable without specifying the relative path.
E.g. type on the command line:
`opa eval "1*2+3"`

If that works, the ssikit should be able to find it and execute it correctly.

By the way, the ssikit docker image contains the opa executable. So using docker, it should just work out of the box.
br
Severin
